### PR TITLE
Turned the $insert variable into a function

### PR DESCRIPTION
**Pull request by marcusxavier85**

### CODE REVIEW

File: insertPokemon.php [hunk 1]
```diff
@@ -2,15 +2,6 @@
...
-
-$pokedexNumber = $_POST['addPokedexNo'];
-$pokemonName = $_POST['addPokemonName'];
-$pokemonType = $_POST['addPokemonType'];
-$pokemonImage = $_POST['addPokemonImage'];
-
-$insert = $dataBase->prepare('INSERT INTO `kanto_pokedex` (`pokedex_no`, `name`, `type`, `img_source`) VALUES (?, ?, ?, ?);');
-
-$insert->execute([$pokedexNumber, $pokemonName, $pokemonType, $pokemonImage]);
+addPokemonToDatabase();
```
Comment (davin2020) [inline insertPokemon.php:4]: I suggest that you check the return type of addPokemonToDatabase() and only forward the user onto the page pokedexDataBaseProject.php if its true ie if you successfully inserted something into the DB

File: functions.php [hunk 2]
```diff
@@ -56,4 +56,22 @@ function displayPokemon(array $pokemons): string {
         }
     }
     return $result;
+}
+
+
+function addPokemonToDatabase() {
+
```
Comment (davin2020) [inline functions.php]: Probably best to remove all the blank lines in between your code, for PSR2

File: functions.php [hunk 3]
```diff
@@ -56,4 +56,22 @@ function displayPokemon(array $pokemons): string {
...
+    $pokedexNumber = $_POST['addPokedexNo'];
+    $pokemonName = $_POST['addPokemonName'];
+    $pokemonType = $_POST['addPokemonType'];
+    $pokemonImage = $_POST['addPokemonImage'];
+
+    $insert = $dataBase->prepare('INSERT INTO `kanto_pokedex` (`pokedex_no`, `name`, `type`, `img_source`) VALUES (?, ?, ?, ?);');
+
+    $insert->execute([$pokedexNumber, $pokemonName, $pokemonType, $pokemonImage]);
+
```
Comment (davin2020) [inline functions.php]: Remove blank lines etc

File: functions.php [hunk 4]
```diff
@@ -56,4 +56,22 @@ function displayPokemon(array $pokemons): string {
         }
     }
     return $result;
+}
+
+
+function addPokemonToDatabase() {
```
Comment (davin2020) [inline functions.php]: For PSR2 and functions the { need to go on its own line ie
`function addPokemonToDatabase() `
`{`

**Review by davin2020 — CHANGES_REQUESTED** [review]

File: insertPokemon.php (hunk 1 above)
Comment (marcusxavier85) [reply insertPokemon.php:4]: no need as the input requires set values to be inputted and something can only be submitted once those values have all been completed

File: functions.php (hunk 2 above)
Comment (marcusxavier85) [reply functions.php]: Removed.

File: functions.php (hunk 3 above)
Comment (marcusxavier85) [reply functions.php]: Removed.

File: functions.php (hunk 4 above)
Comment (marcusxavier85) [reply functions.php]: I thought that was for classes only? As all the funcitons I have written have all been written like that. As in function () {

Comment (davin2020) [reply functions.php]: From PSR website
image

File: insertPokemon.php (hunk 1 above)
Comment (davin2020) [reply insertPokemon.php:4]: Mike suggested its best practise, and that as a user i might expect to see a 'Success' message after submitting a new entry or a Failure/Try Again message if the DB didnt like the submitted values

**Review by davin2020 — CHANGES_REQUESTED** [review]

**Review by davin2020 — APPROVED** [review]

**Pull request merged**